frontend/src/bundles/common/components/video-modal/components/video-modal-content/components/generate-script-form/generate-script-form.tsx

```diff
@@ -27,6 +40,18 @@ const GenerateScriptForm: React.FC<Properties> = ({ onSubmit }) => {
 
     const { handleSubmit, values } = form;
 
+    const isTextGenerated: boolean = useMemo(() => {
+        return messages && messages.length > 0;
+    }, [messages]);
+
+    const getButtonLabel = useCallback(() => {
+        if (dataStatus === DataStatus.PENDING) {
+            return 'Stop';
+        }
+
+        return isTextGenerated ? 'Re-Generate' : 'Generate script';
```

Comment:
```suggestion
        return messages?.length > 0 ? 'Re-Generate' : 'Generate script';
```
I guess you can use this condition directly in this fn because you only use it here. So there is no need for isTextGenerated